client/packages/system/src/Manage/Preferences/EditPage/EditPreference.tsx

```diff
@@ -1,58 +1,66 @@
 import React from 'react';
 import {
-  BasicSpinner,
   Box,
   DetailPanelSection,
+  noOtherVariants,
   NothingHere,
-  PreferenceDescriptionNode,
-  PreferencesNode,
-  usePreferences,
+  PreferenceValueNodeType,
+  Switch,
   useTranslation,
+  isBoolean,
+  isNumber,
+  PreferencesNode,
+  UpsertPreferencesInput,
 } from '@openmsupply-client/common';
-import { EditField } from './EditField';
-import { useEditPreference } from '../api/useEditPreference';
+import { ClientPrefKey } from './getPrefKey';
 
 export const EditPreference = ({
-  selected,
+  valueType,
+  clientKey,
+  value,
+  update,
 }: {
-  selected: PreferenceDescriptionNode;
+  valueType: PreferenceValueNodeType;
+  clientKey: ClientPrefKey;
+  value: PreferencesNode[ClientPrefKey];
+  update: (input: Partial<UpsertPreferencesInput>) => void;
 }) => {
   const t = useTranslation();
 
-  const { data: prefs, isLoading } = usePreferences();
-  const { mutateAsync: update } = useEditPreference(selected.key);
-
-  if (isLoading) {
-    return <BasicSpinner />;
-  }
+  const getRenderer = () => {
+    switch (valueType) {
+      case PreferenceValueNodeType.Boolean:
```

Comment:
This is quite similar to how we went with asset properties - we have a known set of preference value types, and how to render them. Currently, those are scalar types (i.e. bool) but could also include custom ones (e.g. a special renderer for managing custom translation overrides

Reply:
I think would only be defined as a plugin if the pref was also plugged in - otherwise I was just imagining another folder of components here?